core/src/test/java/kafka/test/ClusterInstance.java

```diff
@@ -145,4 +152,21 @@ default Admin createAdminClient() {
     void startBroker(int brokerId);
 
     void waitForReadyBrokers() throws InterruptedException;
+
+    default Set<GroupProtocol> supportedGroupProtocols() {
```

Comment:
It would be great to remove unmodified collection. Also, we need to add comments to explain the reason of checking two configs.
```java
    default Set<GroupProtocol> supportedGroupProtocols() {
        Set<GroupProtocol> supportedGroupProtocols = new HashSet<>();
        supportedGroupProtocols.add(CLASSIC);
        // KafkaConfig#isNewGroupCoordinatorEnabled check both NEW_GROUP_COORDINATOR_ENABLE_CONFIG and GROUP_COORDINATOR_REBALANCE_PROTOCOLS_CONFIG
        if (config().serverProperties().getOrDefault(NEW_GROUP_COORDINATOR_ENABLE_CONFIG, "").equals("true") ||
            config().serverProperties().getOrDefault(GROUP_COORDINATOR_REBALANCE_PROTOCOLS_CONFIG, "").contains("consumer")) {
            supportedGroupProtocols.add(CONSUMER);
        }
        return Collections.unmodifiableSet(supportedGroupProtocols);
    }
```